int-tests/server_controller.py

```diff
@@ -1,21 +1,49 @@
 import subprocess
-import os.path
 import sys
 import functools
+import os.path as path
+from threading import Thread
+from queue import Queue, Empty
 
-_module_dir = os.path.abspath(os.path.join(os.path.dirname(__file__)))
-_root_dir = os.path.abspath(os.path.join(_module_dir, '..'))
+module_dir = path.abspath(path.join(path.dirname(__file__)))
+_root_dir = path.abspath(path.join(module_dir, '..'))
 
 
+class StdOutReader:
+  def __init__(self, stream, verbose=False):
+    self._stream = stream
+    self._queue = Queue()
+    self._verbose = verbose
+
+    def _reader(s, queue):
+      while True:
+        line = s.readline()
+        s.flush()
+        if line:
+          queue.put(line)
+
+    self._thread = Thread(target=_reader, args=(self._stream, self._queue))
+    self._thread.daemon = True
+    self._thread.start()
+
+  def readline(self):
+    try:
+      line = str(self._queue.get(block=False, timeout=0.1))
+      if line and self._verbose:
+        print(line)
+      return line
+    except Empty:
+      return ''
+
```

Comment:
maybe we think of using healthchecks to see if the test servers are up in the futgure?